src/tools/crawler/main.rs

```diff
@@ -75,10 +86,34 @@ fn start_logger(default_level: LevelFilter) {
 async fn main() {
     start_logger(LevelFilter::INFO);
     let args = Args::parse();
+    let mut seed_addrs: Vec<SocketAddr> = Vec::new();
 
     // Create the crawler with the given listener address.
     let crawler = Crawler::new().await;
 
+    if let Some(seeds) = args.seed_addrs {
+        seed_addrs.extend_from_slice(&seeds);
+    }
+
+    if let Some(dns_seed) = args.dns_seed {
+        for seed_str in dns_seed {
+            let response = lookup_host(&seed_str);
+
+            if let Ok(response) = response {
+                for address in response.iter() {
+                    seed_addrs.push(SocketAddr::new(*address, ZCASH_P2P_PORT)); // DNS addrs use this port
+                    info!(parent: crawler.node().span(), "DNS seed {} address added: {}", seed_str, address);
+                }
+            } else {
+                error!(parent: crawler.node().span(), "DNS seed {} lookup failed: {}", seed_str, response.err().unwrap());
+            }
+        }
+    }
+
+    if seed_addrs.is_empty() {
+        panic!("Every seed address failed to resolve. Exiting.");
```

Comment:
I guess this should never happen
```suggestion
        panic!("seed address not found");
```

Reply:
Done.